blockchain/state_manager/src/utils.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use crate::errors::*;
+use crate::StateManager;
+use actor::miner::State;
+use address::{Address, Payload};
+use blockstore::BlockStore;
+use cid::Cid;
+use fil_types::{RegisteredProof, SectorInfo, SectorSize};
+use filecoin_proofs_api::{post::generate_winning_post_sector_challenge, ProverId};
+use std::convert::TryInto;
+
+pub fn get_sectors_winning_for_winning_post<DB>(
+    state_manager: &StateManager<DB>,
+    st: &Cid,
+    address: &Address,
+    rand: &[u8; 32],
+) -> Result<Vec<SectorInfo>, Error>
+where
+    DB: BlockStore,
+{
+    let miner_actor_state: State = state_manager.load_actor_state(&address, &st)?;
+    let mut not_proving = miner_actor_state
+        .faults
+        .clone()
+        .merge(&miner_actor_state.recoveries)
+        .map_err(|_| Error::Other("Could not merge bitfield".to_string()))?;
+    let sector_set =
+        miner_actor_state.load_sector_infos(&*state_manager.get_block_store(), &mut not_proving)?;
```

Comment:
If matching lotus with this, might as well put it into it's own function so it can be reused later